client/src/components/Modal/ModalHeader.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import i18n from 'i18n';
+import { ModalHeader as ReactStrapModalHeader } from 'reactstrap';
+import PropTypes from 'prop-types';
+
+/**
+ * Component to render a modal header.
+ */
+function ModalHeader({ onClosed, title, ModalHeaderComponent, showCloseButton }) {
+  if (title !== false) {
+    if (typeof title === 'object') {
+      // FormSchema title occasionally contains html, only render text for modal title
+      const doc = new DOMParser().parseFromString(title.html, 'text/html');
+      title = doc.body.textContent || '';
+    }
+    return (
+      <ModalHeaderComponent toggle={onClosed}>{title}</ModalHeaderComponent>
+    );
+  }
+
+  if (showCloseButton === true && typeof onClosed === 'function') {
```

Comment:
```suggestion
  if (showCloseButton && onClosed) {
```
Rely on propTypes for type checking

Reply:
See https://github.com/silverstripe/silverstripe-admin/pull/1631#discussion_r1487142355